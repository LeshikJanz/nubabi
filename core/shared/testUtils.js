import configureMockStore from 'redux-mock-store';
import { combineEpics, createEpicMiddleware } from 'redux-observable';

export const setupEpic = (epic) => {
  const epicMiddleware = createEpicMiddleware(epic);
  const mockStore = configureMockStore([epicMiddleware]);

  return {
    middleware: epicMiddleware,
    mockStore: mockStore,
    afterEach: () => {
      epicMiddleware.replaceEpic(epic);
    }
  }
};
