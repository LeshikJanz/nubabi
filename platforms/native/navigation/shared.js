import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) =>
  pathOr('Nubabi', ['params', 'babyName'], state);

export function getBabyTitle(navigation) {
  if (!navigation) {
    throw new Error('an screen hasnt been updated, check stack');
  }

  return babyNameTitle(navigation);
}

export const getTabHeaders = navigate => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
