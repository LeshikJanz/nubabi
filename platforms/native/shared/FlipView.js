// @flow
import React, { Component } from 'react';
import { View, Animated } from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

type Props = {
  renderFront: Function,
  renderBack: Function,
  velocity: number,
  tension: number,
  friction: number,
};

export class FlipCard extends Component {
  props: Props;

  state = {
    animatedValue: new Animated.Value(0),
    isFlipped: true,
  };

  componentDidUpdate(prevProps: Props, prevState: *) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard = () => {
    this.setState({ isFlipped: !this.state.isFlipped });
  };

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0, // Returns to the start
      velocity: this.props.velocity, // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction, // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront(this._flipToggleCard);
    }
    return this.props.renderBack(this._flipToggleCard);
  }

  render() {
    const perspective = 2000;
    const rotateY = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <View style={styles.animatedContainer}>
        <Animated.View
          style={[
            styles.animatedContainer,
            { transform: [{ rotateY }, { perspective }] },
          ]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </View>
    );
  }
}

export default FlipCard;
