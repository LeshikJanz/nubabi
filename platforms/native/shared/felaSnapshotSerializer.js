// WIP
// prettier-ignore
const felaSnapshotSerializer = {
  test(val) {
    return val && val.felaRules && val.$$typeof === Symbol.for('react.test.json');
  },

  print(val, print) {
    const styles = val.felaRules;
    val.felaRules = false;

    return `style = Object ${JSON.stringify(styles, null, 4)}\n\n${print(val)}`;
  },
};

module.exports = felaSnapshotSerializer;
