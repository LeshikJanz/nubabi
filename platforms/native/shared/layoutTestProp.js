export default {
  fontScale: 2,
  height: 1334,
  scale: 2,
  width: 750,
  viewportWidth: 750,
  viewportHeight: 1334,
};
