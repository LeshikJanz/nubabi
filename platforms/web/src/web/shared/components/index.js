import About from "./About";
import AsyncComponent from "./AsyncComponent";
import AuthenticatedRoute from "./AuthenticatedRoute";
import Header from "./Header";
import Home from "./Home";
import Loader from "./Loader";
import Test from "./Test";

export default {
  About,
  AsyncComponent,
  AuthenticatedRoute,
  Header,
  Home,
  Loader,
  Test
};
