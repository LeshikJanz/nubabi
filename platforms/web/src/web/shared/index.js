import * as actions from "./actions";
export { default as containers } from "./containers";
export { default as reducer } from "./reducer";
export { default as epics } from "./epics";
export { actions };
