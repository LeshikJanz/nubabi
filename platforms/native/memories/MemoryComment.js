// @flow
import type { Comment } from 'core/types';
import React from 'react';
import { path } from 'ramda';
import { gql } from 'react-apollo';
import moment from 'moment';
import { Avatar, Box, Text } from '../components';

type Props = Comment & {};

export const MemoryComment = ({ text, author, createdAt }: Props) => {
  const timestamp = moment(createdAt);

  let format = 'D MMMM HH:mm';

  if (timestamp.year() !== moment().year()) {
    format = 'D MMMM YYYY';
  }

  const metadataText = [
    author.firstName,
    author.lastName,
    'â€¢',
    timestamp.format(format),
  ].join(' ');

  const avatar = path(['avatar', 'url'], author);

  return (
    <Box contentSpacing flexDirection="row">
      <Box
        style={() => ({ marginRight: 10 })}
        alignItems="center"
        marginTop={0.5}
        justifyContent="flex-start"
      >
        <Avatar src={avatar} size={30} />
      </Box>
      <Box flex={1}>
        <Box>
          <Text
            style={() => ({
              letterSpacing: 0.17,
              lineHeight: 20,
              fontSize: 14,
              marginRight: 16,
            })}
          >
            {text}
          </Text>
        </Box>
        <Box marginTop={0.5}>
          <Text color="secondary">{metadataText}</Text>
        </Box>
      </Box>
    </Box>
  );
};

MemoryComment.fragments = {
  comment: gql`
    fragment MemoryComment on Comment {
      id
      text
      createdAt
      author {
        id
        firstName
        lastName
        avatar {
          url
        }
      }
    }
  `,
};

export default MemoryComment;
