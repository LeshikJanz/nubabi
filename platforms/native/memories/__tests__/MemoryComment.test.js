import 'react-native';
import React from 'react';
import MemoryComment from '../MemoryComment';
import { expectRender } from '../../shared/testUtils';

describe('MemoryComment', () => {
  it('renders correctly', () => {
    const comment = {
      id: 1,
      text: 'Some comment',
      createdAt: new Date(2017, 8, 8),
      author: {
        firstName: 'Alice',
        lastName: 'Doe',
        avatar: {
          url: 'http://example.com',
        },
      },
    };

    expectRender(<MemoryComment {...comment} />);
  });
});
