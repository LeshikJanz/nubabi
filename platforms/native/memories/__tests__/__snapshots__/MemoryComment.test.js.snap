// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemoryComment renders correctly 1`] = `
style = Object {
    "paddingBottom": 10,
    "paddingLeft": 10,
    "paddingRight": 10,
    "paddingTop": 10,
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "flexDirection": "row",
    "marginTop": 8,
    "alignItems": "center",
    "justifyContent": "center",
    "marginRight": 16,
    "backgroundColor": "#FFFFFF",
    "borderRadius": 15,
    "height": 30,
    "width": 30,
    "overflow": "hidden",
    "flex": 1,
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#748294",
    "fontFamily": "SF Pro Text",
    "letterSpacing": 0.17
}

<View
  contentSpacing={true}
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "justifyContent": "flex-start",
        "marginRight": 10,
        "marginTop": 8,
      }
    }
  >
    <View
      id={undefined}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 15,
          "height": 30,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 30,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://example.com",
          }
        }
        style={
          Object {
            "flex": 1,
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#9EABBC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "color": "#454D56",
            "fontFamily": "SF Pro Text",
            "fontSize": 14,
            "letterSpacing": 0.17,
            "lineHeight": 20,
            "marginRight": 16,
          }
        }
      >
        Some comment
      </Text>
    </View>
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "marginTop": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#9EABBC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "color": "#748294",
            "fontFamily": "SF Pro Text",
            "fontSize": 13,
            "lineHeight": 16,
          }
        }
      >
        Alice Doe â€¢ 8 September 00:00
      </Text>
    </View>
  </View>
</View>
`;
